### PR TITLE
Emulate how eigen handles status bars

### PR DESCRIPTION
**Pull request by orta**
Now the status bar is provided at the root level of the VC hierarchy, so individual screens shouldn't have to care about it at all.

screen shot 2018-01-22 at 4 42 00 pm

Previous to this PR, there wouldn't be that white bar.

### CODE REVIEW

File: src/lib/Scenes/Home/index.tsx [hunk 1]
```diff
@@ -19,9 +19,7 @@ import renderWithLoadProgress from "lib/utils/renderWithLoadProgress"
 import DarkNavigationButton from "lib/Components/Buttons/DarkNavigationButton"
 import TabBar, { Tab } from "lib/Components/TabBar"
 
-const TabBarContainer = styled.View`
-  margin-top: 20px;
-`
+const TabBarContainer = styled.View``
```
Comment (orta) [inline src/lib/Scenes/Home/index.tsx]: This is the only real change in user-land code.

File: Example/Emission/ARTopMenuViewController.m [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+- (void)viewDidLoad
+{
+  [super viewDidLoad];
+
+  _statusBarView = [[UIView alloc] init];
+  _statusBarView.backgroundColor = UIColor.whiteColor;
+
+  [self.view addSubview:_statusBarView];
+
+  _statusBarVerticalConstraint = [_statusBarView constrainHeight:@"20"];
```
Comment (ashfurrow) [inline Example/Emission/ARTopMenuViewController.m]: Shouldn't the `20` here be derived from calling `statusBarHeight`?

File: Example/Emission/ARTopMenuViewController.m [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+  }
+}
+
+- (void)viewDidLayoutSubviews
+{
+  [super viewDidLayoutSubviews];
+
+  if (_statusBarVerticalConstraint.constant != [self statusBarHeight]) {
+    _statusBarVerticalConstraint.constant = [self statusBarHeight];
+  }
```
Comment (ashfurrow) [inline Example/Emission/ARTopMenuViewController.m]: Why is this necessary? Could be a helpful comment.

**Comment by orta** [comment]
Done - thanks 👍

File: Example/Emission/AppDelegate.m [hunk 4]
```diff
@@ -64,7 +65,7 @@ - (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(
 
   self.window = [[UIWindow alloc] initWithFrame:[[UIScreen mainScreen] bounds]];
   self.window.backgroundColor = [UIColor whiteColor];
-  self.window.rootViewController = self.navigationController;
+  self.window.rootViewController = [[ARTopMenuViewController alloc] initWithNavigationController:self.navigationController];;
```
Comment (alloy) [inline Example/Emission/AppDelegate.m]: Needs more semi-colons! 🔔
